lib/api/core/storage/storageEngine.js

```diff
@@ -22,67 +22,71 @@
 'use strict';
 
 const
-  assert = require('assert'),
-  errorsManager = require('../../../config/error-codes/throw'),
+  Elasticsearch = require('../../../services/elasticsearch'),
```

Comment:
New StorageEngine class who instantiate the `public` and `internal` storage engines through `ClientAdapter` class.  
This class also keep the RAM index cache